punchcut/src/outline.rs

```diff
@@ -0,0 +1,293 @@
+/*!
+Representation of a simple outline glyph.
+*/
+
+use peniko::kurbo::{
+    segments, BezPath, ParamCurveArclen, ParamCurveArea, ParamCurveExtrema, PathEl, Point, Rect,
+    Shape,
+};
+
+use crate::source::*;
+
+/// Simple outline represented as a sequence of path commands.
+#[derive(Clone, Default, Debug)]
+pub struct Outline {
+    verbs: Vec<Verb>,
+    points: Vec<(f32, f32)>,
```

Comment:
Is this another case where we wish we had 32-bit kurbo? I definitely think some of the code below would be easier to follow if the math was implemented in terms of a `Point` type...

Reply:
Yes, `PathEl` is 56 bytes while the typical quadratic segment can be represented in 17 bytes in this representation. This is a fairly significant difference. From the most recent discussion, we may remove this entirely though and just add a path sink style trait that would avoid the issue.